npm/src/clientfileextractmetadata/index.ts

```diff
@@ -0,0 +1,16 @@
+import {
+  extractFileMetadata,
+  IFileMetadata,
+  TdrFile
+} from "@nationalarchives/file-information"
+
+export class ClientFileExtractMetadata {
+  async extract(files: TdrFile[]): Promise<IFileMetadata[]> {
+    try {
+      const metadata: IFileMetadata[] = await extractFileMetadata(files)
+      return metadata
```

Comment:
There's the option to pass a callback here. I think we're going to show the progress of the metadata extraction as part of the single progress bar so we'll need a callback. I've got an idea for it actually that I can re-use for my s3 upload stuff. I'll call you about it in the morning.

Reply:
As per separate conversation, have added a temporary call back function  until s3 upload functionality is in place.